zstacklib/zstacklib/utils/gpu.py

```
@@ -226,6 +226,55 @@ def get_huawei_gpu_product_name_cmd(npu_id, iswindows=False):
     return cmd
 
 
+def get_huawei_gpu_aios_rank_table_dict(npu_ids, iswindows=False):
+    for npu_id in npu_ids:
+        if not npu_id.isdigit():
+            raise ValueError("NPU ID must be a digit, got: {}".format(npu_id))
+
+    # Build the command to get IP addresses for each NPU ID using hccn_tool
+    device_ips = {}
+    for npu_id in npu_ids:
+        r, o, e = bash_roe(f"hccn_tool -i {npu_id} -ip -g")
```

Ruff (0.12.2)

`bash_roe` may be undefined, or defined from star imports

(F405)